src/Services/Git/GitonomyClient.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+namespace fortrabbit\Copy\Services\Git;
+
+use fortrabbit\Copy\Exceptions\GitException;
+use Gitonomy\Git\Admin;
+use Gitonomy\Git\Exception\ProcessException;
+use Gitonomy\Git\Reference\Branch;
+use Gitonomy\Git\Repository;
+use LogicException;
+use RuntimeException;
+
+class GitonomyClient implements Client
+{
+
+	private Repository $repository;
+
+	private string $directory;
+
+	public function clone(
+		string $repository,
+		?string $directory = null,
+		array $options = []
+	) {
+		$this->directory = $directory;
+		$this->repository = Admin::cloneTo($directory, $repository, true, $options);
+	}
+
+	public function push(string $upstream, string $branch = 'master'): string
+	{
+		return $this->run(GitCommand::PUSH, [$upstream, $branch]);
+	}
+
+	public function pull(string $upstream, string $branch = 'master'): string
+	{
+		return $this->run(GitCommand::PULL, [$upstream, $branch]);
+	}
+
+	public function getLocalHead(): ?string
+	{
+		$head = $this->repository->getHead();
+		if ($head instanceof Branch) {
+			return $head->getName();
+		}
+		return $head->getFullName();
+	}
+
+	public function getLocalBranches(): array
+	{
+		$localBranches = array_filter(
+			$this->repository->getReferences()->getBranches(),
+			fn ($branch) => $branch->isLocal()
+		);
+
+		return array_map(function ($branch) {
+			return $branch->getName();
+		}, $localBranches);
+	}
+
+	public function getRemotes(?string $for = 'push'): array
+	{
+		if (!in_array($for, ['push', 'pull'], true)) {
+			throw new LogicException(
+				sprintf(
+					'Argument 1 passed to %s must be "pull" or "push", %s given.',
+					'fortrabbit\Copy\Services\Git\GitonomyClient::getRemotes()',
+					$for
+				)
+			);
+		}
+
+		$remotes = explode(PHP_EOL, rtrim($this->run(GitCommand::REMOTE, ['-v'])));
+
+		return $remotes;
+	}
+
+	public function getTracking(bool $includeBranch = false): ?string
+	{
+		try {
+			$result = $this->repository->run(GitCommand::REV_PARSE, [
+				'--symbolic-full-name',
+				'@{u}',
+				'--abbrev-ref',
+			]);
+		} catch (ProcessException) {
+			return null;
+		}
+
+		if ($includeBranch) {
+			return $result;
+		}
+
+		// Split upstream/branch and return upstream only
+		return explode('/', $result)[0];
+	}
+
+	public function checkout(string $branch)
+	{
+		$this->repository->run(GitCommand::CHECKOUT, [$branch]);
+	}
+
+	public function addRemote(string $name, ?string $url)
```

Comment:
@pheeque1 
I tested it and notice the remote was not added.
That's the reason why :-)

Reply:
🤦🏾‍♂️ Totally missed this. Implemented now.
Thanks.